stdlib/public/core/ArrayBufferType.swift

```diff
@@ -157,28 +157,31 @@ extension _ArrayBufferType {
       newTailStart.moveInitializeBackwardFrom(oldTailStart, count: tailCount)
 
       // Assign over the original subRange
-      var i = newValues.startIndex
-      for j in subRange {
-        elements[j] = newValues[i]
-        i._successorInPlace()
+      var generator = newValues.generate()
+      for i in subRange { 
+        let element = generator.next()
+        _collectionAssert(element != nil)
```

Comment:
This assertion seems like it’s going to slow down non-optimized code just to get a better error message than the force-unwrap would supply.  Is it worth it?